### ISSUE TITLE
Calling sandbox.replace not cleaning up if called more than once

### ISSUE THREAD

**Issue by mantoni** [sinonjs/sinon#1779]
When calling `sandbox.replace` twice for the same `object` + `property`, the original state can not be restored.

Reproduce:

```js
sinon.replace(console, "log", sinon.fake());
sinon.replace(console, "log", sinon.fake());
sinon.restore();
console.log; // Still a proxy object
```

There are two different ways to solve this:

- We throw on the second attempt to use `sinon.replace`, like `sinon.stub(obj, prop)` fails if it's already a fake
- We allow multiple calls to use `sinon.replace`, but restore to the original state properly

I'd try to come up with an implementation if we know which direction to take. I'm not really decided here.

**Comment by mroderick** [comment]
Trying to replace a fake with a fake seems like something that should warrant an investigation by the programmer. I would prefer to have tests fail, rather than sinon glossing over my mistakes. Throwing get's my vote.

**Comment by mroderick** [comment]
This was fixed with `sinon@5.0.5`
